seacrowd/sea_datasets/uit_visfd/uit_visfd.py

```diff
@@ -0,0 +1,231 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Crawled textual feedback from customers about smartphones on \
+    a large e-commerce website in Vietnam. The label of the \
+    dataset is ten aspects and three polarities. Please read \
+    the guidelines in the paper for more information. \
+    We randomly divide the dataset into three sets:
+"""
+import os
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import datasets
+import pandas as pd
+
+from seacrowd.utils import schemas
+from seacrowd.utils.configs import SEACrowdConfig
+from seacrowd.utils.constants import Licenses, Tasks
+
+# TODO: Add BibTeX citation
+_CITATION = """\
+  @InProceedings{10.1007/978-3-030-82147-0_53,
+  author="Luc Phan, Luong
+  and Huynh Pham, Phuc
+  and Thi-Thanh Nguyen, Kim
+  and Khai Huynh, Sieu
+  and Thi Nguyen, Tham
+  and Thanh Nguyen, Luan
+  and Van Huynh, Tin
+  and Van Nguyen, Kiet",
+  editor="Qiu, Han
+  and Zhang, Cheng
+  and Fei, Zongming
+  and Qiu, Meikang
+  and Kung, Sun-Yuan",
+  title="SA2SL: From Aspect-Based Sentiment Analysis to Social Listening System for Business Intelligence",
+  booktitle="Knowledge Science, Engineering and Management ",
+  year="2021",
+  publisher="Springer International Publishing",
+  address="Cham",
+  pages="647--658",
+  isbn="978-3-030-82147-0"
+  }
+"""
+
+_DATASETNAME = "uit_visfd"
+
+_DESCRIPTION = """
+UIT-ViSFD is the Vietnamese Smartphone Feedback Dataset.
+It is an aspect-based sentiment analysis dataset.
+It consists of 11,122 human-annotated comments for mobile e-commerce.
+"""
+
+_HOMEPAGE = "https://github.com/LuongPhan/UIT-ViSFD"
+
+_LANGUAGES = ["vie"]
+
+_LICENSE = Licenses.UNKNOWN.value
+
+_LOCAL = False
+
+_URLS = {_DATASETNAME: "https://github.com/LuongPhan/UIT-ViSFD/raw/main/UIT-ViSFD.zip"}
+
+_SUPPORTED_TASKS = [Tasks.ASPECT_BASED_SENTIMENT_ANALYSIS]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class UITViSFDDataset(datasets.GeneratorBasedBuilder):
+    """
+    Crawled textual feedback from customers about smartphones on a large e-commerce website in Vietnam.
+    The label of the dataset is ten aspects and three polarities.
+    Please read the guidelines in the paper for more information.
+    We randomly divide the dataset into three sets:
+        - Train: 7,786.
+        - Dev: 1,112.
+        - Test: 2,224.
+    """
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+    SEACROWD_SCHEMA_NAME = "text_multi"
```

Comment:
There are dataloaders that support multiple schemas, i.e.: datasets with multiple tasks.
Hence, this is not a recommended practice.
Could you please remove this?

Reply:
I guess we can't remove `SOURCE_VERSION` and `SEACROWD_VERSION` Sir, because they will be used on `BUILDER_CONFIGS`.

Do you want me to remove `SEACROWD_SCHEMA_NAME` constant?

Reply:
I just try to be consistent with other approved-dataloader on the `main` branch, such as:
- [asr_stidusc](https://github.com/SEACrowd/seacrowd-datahub/blob/42d6285b75583e3ac932c2ac0e86b8fa3d6f6be7/seacrowd/sea_datasets/asr_stidusc/asr_stidusc.py#L43-L55)
- [bhinneka_korpus](https://github.com/SEACrowd/seacrowd-datahub/blob/42d6285b75583e3ac932c2ac0e86b8fa3d6f6be7/seacrowd/sea_datasets/bhinneka_korpus/bhinneka_korpus.py#L33-L52)
- [burmese_romanize](https://github.com/SEACrowd/seacrowd-datahub/blob/42d6285b75583e3ac932c2ac0e86b8fa3d6f6be7/seacrowd/sea_datasets/burmese_romanize/burmese_romanize.py#L39-L50)
- [cc_aligned_doc](https://github.com/SEACrowd/seacrowd-datahub/blob/42d6285b75583e3ac932c2ac0e86b8fa3d6f6be7/seacrowd/sea_datasets/cc_aligned_doc/cc_aligned_doc.py#L60-L71)
- [cc_aligned_sent](https://github.com/SEACrowd/seacrowd-datahub/blob/42d6285b75583e3ac932c2ac0e86b8fa3d6f6be7/seacrowd/sea_datasets/cc_aligned_sent/cc_aligned_sent.py#L61-L74)
- etc.

Those dataloader have implemented `SEACROWD_SCHEMA_NAME` for just one `Tasks` only.

Based on my consideration above, do you still recommend to remove `SEACROWD_SCHEMA_NAME`, Sir?

Reply:
yes, the `SEACROWD_SCHEMA_NAME`.

Do note that this thread is only for line 98, github provides a few lines back in snippet only to get better context.
When the thread is meant for multiple lines, it shows additional text as attached
![image](https://github.com/SEACrowd/seacrowd-datahub/assets/6541786/8eae05e0-d055-48b5-9bd5-8b91f73f42d9)

Reply:
Oh, I see, Sir. I just know it now that Github gives context on the conversation :open_mouth:.

Reply:
So, for the usage of `SEACROWD_SCHEMA_NAME`, do you still recommend removing it Sir? I will remove it ASAP if you want me to.

Reply:
yes.

Reply:
Alright, solved on https://github.com/SEACrowd/seacrowd-datahub/pull/637/commits/4433e1dcf0c6fcbbd4fdf28311292c88f296c3ce, Sir.